### ISSUE TITLE
[diagnostic] warn on `typeof` undeclared variable 

### ISSUE THREAD

**Issue by vegerot** [quick-lint/quick-lint-js#825]
While working on my first JS project in a while, I wrote this bug

```js
function foo(obj1, obj2) {
if (typeof obj !== obj2) {…}
…
}
```
and was disappointed quick-lint didn’t error or at least give me a warning

**Comment by strager** [comment]
`typeof` is often used to detect if a variable exists in the global scope. We can't *know* what the programmer intended for `typeof` in your example.

Can you think of a heuristic which will catch your example but not produce false positive warnings for things like `typeof WebTransport`?

**Comment by CoderMuffin** [comment]
@strager maybe it should be a warning unless it is immediately compared to "undefined"?

**Comment by CoderMuffin** [comment]
@strager maybe it should be a warning unless it is immediately compared to "undefined"?
maybe a config setting could be added to disable it if it is too aggressive
